demo-app/src/App-NonRedux.tsx

```diff
@@ -14,52 +15,113 @@ const actionMapper: {[key: string]: (node: NodeProps, value: any) => NodeProps}
     [ActionTypes.DISABLED]: Tree.nodeDisabled,
     [ActionTypes.SELECTED]: Tree.nodeSelected,
     [ActionTypes.CHILD_NODES]: Tree.nodeChildren,
-    [ActionTypes.LOADING]: Tree.nodeLoading,
+    [ActionTypes.LOADING]: Tree.nodeLoading
 };
 
 export class AppNonRedux extends React.Component<{}, AppState> {
     state = {
-        tree: Tree.convertHierarchicalTree(Tree.initHierarchicalTree(generator()))
+        tree: Tree.convertHierarchicalTree(Tree.initHierarchicalTree(generator())),
+        commandHistory: [] as (CommandQueueType & {key: Number})[],
     };
 
+    // This is needed. At too fast changes (multiple changes before the state changes)
```

Comment:
@skateman I was not able to do better, I am open to ideas how to solve this problem without an extra internal variable. I don't like it.

Reply:
I don't mind if this is just the demo, but @karelhala @Hyperkid123  will tell you more